factory-method/src/main/java/com/iluwatar/factory/method/ElfBlacksmith.java

```diff
@@ -22,16 +22,25 @@
  */
 package com.iluwatar.factory.method;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * 
  * Concrete subclass for creating new objects.
  * 
  */
 public class ElfBlacksmith implements Blacksmith {
 
+  private static Map<WeaponType, ElfWeapon> ELF_ARSENAL = new HashMap<>(WeaponType.values().length);
+  {
```

Comment:
This should be a static initializer block, so that initialization is done once only and not for every class instance